docs/api.md

```diff
@@ -1799,6 +1804,20 @@ puppeteer.launch().then(async browser => {
 
 In the case of multiple pages in a single browser, each page can have its own viewport size.
 
+To avoid race condition with page's logic that sniffs for viewport properties,
```

Comment:
page.setViewport will resize the page. A lot of websites don't expect phones to change size, so you should set the viewport before navigating to the page.